tests/test_popstepsampling.py

```diff
@@ -54,6 +54,25 @@ def test_stepsampler_cubegausswalk(plot=False):
     assert a.sum() > 1
     assert b.sum() > 1
 
+def test_stepsampler_randomEllSlice(plot=False):
+    np.random.seed(4)
+    nsteps = np.random.randint(10, 50)
+    popsize = np.random.randint(1, 20)
+    sampler = ReactiveNestedSampler(paramnames, loglike_vectorized, transform=transform, vectorized=True)
+
+    sampler.stepsampler = PopulationEllipticalSliceSampler(
+        popsize=popsize, nsteps=nsteps, 
+        generate_direction=generate_random_direction,
+        scale=1.0,
+    )
+    r = sampler.run(viz_callback=None, log_interval=50, max_iters=200, max_num_improvement_loops=0)
+    sampler.print_results()
+    a = (np.abs(r['samples'] - 0.7) < 0.1).all(axis=1)
```

Comment:
this tests at least that nothing is crashing and that the sampling progresses towards the MLE. It would be better if any logic of PopulationEllipticalSliceSampler can be pulled out into a function or method and tested. For example, run a single iteration of the sampler with the initial seed/state fixed and check that the new state is sane, and what adaptation was performed.